module/applications/dice/roll-configuration-dialog.mjs

```diff
@@ -0,0 +1,295 @@
+import Application5e from "../api/application.mjs";
+
+/**
+ * Dialog rendering options for a roll configuration dialog.
+ *
+ * @typedef {object} BasicRollConfigurationDialogOptions
+ * @property {typeof BasicRoll} rollType  Roll type to use when constructing final roll.
+ * @property {object} [default]
+ * @property {number} [default.rollMode]  Default roll mode to have selected.
+ * @property {*} [resolve]                Method to call when resolving successfully.
+ * @property {*} [reject]                 Method to call when the dialog is closed or process fails.
+ */
+
+/**
+ * Dialog for configuring one or more rolls.
+ *
+ * @param {BasicRollProcessConfiguration} [config={}]          Initial roll configuration.
+ * @param {BasicRollMessageConfiguration} [message={}]        Message configuration.
+ * @param {BasicRollConfigurationDialogOptions} [options={}]  Dialog rendering options.
+ */
+export default class RollConfigurationDialog extends Application5e {
+  constructor(config={}, message={}, options={}) {
+    super(options);
+
+    /**
+     * Roll configuration.
+     * @type {BasicRollProcessConfiguration}
+     */
+    Object.defineProperty(this, "config", { value: config, writable: false });
+
+    this.message = message;
+    this.rolls = this.#buildRolls(foundry.utils.deepClone(this.config));
+  }
+
+  /* -------------------------------------------- */
+
+  /** @override */
+  static DEFAULT_OPTIONS = {
+    classes: ["roll-configuration"],
+    tag: "form",
```

Comment:
My one concern would be players who begin a roll and then want to look up something on their sheet to see if they have advantage or some sort of situational bonus.

It might be good to be a `popover` instead of modal, so it will always stay on top but still allow for interacting with other applications. Unfortunately core's window dragging doesn't seem to interact very well with popovers.

```javascript
  /** @inheritDoc */
  async _onFirstRender(context, options) {
    await super._onFirstRender(context, options);
    this.element.setAttribute("popover", "");
  }
```

Reply:
Yeah that makes sense. Probably more trouble than it's worth at this stage to use popover or dialog.